circuit/types/field/src/equal.rs

```diff
@@ -110,21 +128,6 @@ impl<E: Environment> Equal<Self> for Field<E> {
                 // Check 2:  (a - b) * not(1) = 0
                 //                          0 = 0
                 // => is_neq is trivially correct.
```

Comment:
Is this documentation still relevant/does it need to be updated?

Reply:
It looks like this is still relevant.

It may be advisible to revert the bottom code so that its documentation remains in place.

Reply:
I think it's still relevant and applies in its current form, but we could also consider shortening it instead (I can add a proposed block of comments with a shorter version; let me know).

Reply:
We could move some of the code relative to the block comment, but I've also refactored it slightly, so I'm not sure where exactly we want to move the block comment.

Reply:
It seems we can revert L66-69 (new) back to L118-127 (old).

Reply:
And agreed, do you want to proposed a simplified "proof" in the comments (for the cases)?

Reply:
I just pushed a commit with a shorter explanation of correctness. I also put it before the code, to address the code movement you mentioned (I moved the comment instead). But let me know if you'd like something different.